src/main/java/com/algorand/algosdk/v2/client/indexer/LookupApplicationLogsByID.java

```diff
@@ -0,0 +1,114 @@
+package com.algorand.algosdk.v2.client.indexer;
+
+import com.algorand.algosdk.crypto.Address;
+import com.algorand.algosdk.v2.client.common.Client;
+import com.algorand.algosdk.v2.client.common.HttpMethod;
+import com.algorand.algosdk.v2.client.common.Query;
+import com.algorand.algosdk.v2.client.common.QueryData;
+import com.algorand.algosdk.v2.client.common.Response;
+import com.algorand.algosdk.v2.client.model.ApplicationLogsResponse;
+
+
+/**
+ * Lookup application logs.
+ * /v2/applications/{application-id}/logs
+ */
+public class LookupApplicationLogsByID extends Query {
+
+    private Long applicationId;
+
+    /**
+     * @param applicationId
+     */
+    public LookupApplicationLogsByID(Client client, Long applicationId) {
+        super(client, new HttpMethod("get"));
+        this.applicationId = applicationId;
+    }
+
+    /**
+     * Maximum number of results to return.
+     */
+    public LookupApplicationLogsByID limit(Long limit) {
+        addQuery("limit", String.valueOf(limit));
+        return this;
+    }
+
+    /**
+     * Include results at or before the specified max-round.
+     */
+    public LookupApplicationLogsByID maxRound(Long maxRound) {
+        addQuery("max-round", String.valueOf(maxRound));
+        return this;
+    }
+
+    /**
+     * Include results at or after the specified min-round.
+     */
+    public LookupApplicationLogsByID minRound(Long minRound) {
+        addQuery("min-round", String.valueOf(minRound));
+        return this;
+    }
+
+    /**
+     * The next page of results. Use the next token provided by the previous results.
+     */
+    public LookupApplicationLogsByID next(String next) {
```

Comment:
nit: maybe change next to nextToken to be consistent with the other sdks.

Reply:
I agree but this is generated code, and all query types use `next` for the next token.